doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -659,168 +734,476 @@ insert in a concise way an Ltac2 variable of type :n:`constr` into a Coq term.
-   hypmatching      : `lident` : `constrpattern`
-                    : _ : `constrpattern`
+   .. insertprodn goal_match_list gmatch_hyp_pattern
 
-Variables from :n:`@hypmatching` and :n:`@constrpattern` are bound in the body of the
-branch. Their types are:
+   .. prodn::
+      goal_match_list ::= {? %| } {+| @gmatch_rule }
+      gmatch_rule ::= @gmatch_pattern => @ltac2_expr
+      gmatch_pattern ::= [ {*, @gmatch_hyp_pattern } |- @ltac2_match_pattern ]
+      gmatch_hyp_pattern ::= @name : @ltac2_match_pattern
 
-- ``constr`` for pattern variables appearing in a :n:`@term`
-- ``Pattern.context`` for variables binding a context
-- ``ident`` for variables binding a hypothesis name.
+   Matches over goals, similar to Ltac1 :tacn:`match goal`.
+   Use this form to match hypotheses and/or goals in the proof context.  These patterns have zero or
+   more subpatterns to match hypotheses followed by a subpattern to match the conclusion.  Except for the
+   differences noted below, this works the same as the corresponding :n:`@ltac2_match_key @ltac2_expr` construct
+   (see :tacn:`match!`).  Each current goal is processed independently.
 
-The same identifier caveat as in the case of matching over constr applies, and
-this features has the same semantics as in Ltac1. In particular, a ``reverse``
-flag can be specified to match hypotheses from the more recently introduced to
-the least recently introduced one.
+   :n:`{*, @gmatch_pattern }`
+      Patterns to match with hypotheses.  Each pattern must match a distinct hypothesis in order
+      for the branch to match.
 
-.. _ltac2_notations:
+      Hypotheses have the form :n:`@name {? := @term__binder } : @type`.  Currently Ltac2 doesn't
+      allow matching on or capturing the value of :n:`@term__binder`.  It only supports matching on
+      the :token:`name` and the :token:`type`, for example `n : ?t`.
 
-Notations
----------
+      .. currently only supports the first row
+         :list-table::
+         :widths: 2 1
+         :header-rows: 1
 
-Notations are the crux of the usability of Ltac1. We should be able to recover
-a feeling similar to the old implementation by using and abusing notations.
+         * - Pattern syntax
+           - Example pattern
 
-Scopes
-~~~~~~
+         * - :n:`@name : @ltac2_match_pattern`
+           - `n : ?t`
 
-A scope is a name given to a grammar entry used to produce some Ltac2 expression
-at parsing time. Scopes are described using a form of S-expression.
+         * - :n:`@name := @match_pattern__binder`
+           - `n := ?b`
 
-.. prodn::
-   ltac2_scope ::= {| @string | @int | @lident ({+, @ltac2_scope}) }
+         * - :n:`@name := @term__binder : @type`
+           - `n := ?b : ?t`
 
-A few scopes contain antiquotation features. For the sake of uniformity, all
-antiquotations are introduced by the syntax :n:`$@lident`.
+         * - :n:`@name := [ @match_pattern__binder ] : @ltac2_match_pattern`
```

Comment:
This isn't part of the syntax you note above?  OR am I missing something?

Reply:
Nope.  Compare the statement to the syntax in the grammar.

Reply:
Ah, I believe this is in your comment "currently only supports the first row".  Perhaps we should comment out the other rows?

Reply:
It is commented out.  Rst syntax is like Python, it goes by indentation.  Comments begin with `.. ` and end just before the first following line that is indented the same amount or less.

You may want to look at the HTML output, much easier to read.